qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/completions/QuteCompletionsForExpression.java

```diff
@@ -834,12 +836,33 @@ private void doCompleteExpressionForObjectPartWithParentNodes(Node part, Node no
 							}
 						}
 						break;
+					case WHEN:
+					case SWITCH:
+						// Completion for properties/methods of with object from #switch and #when
+						Parameter value = ((BaseWhenSection) section).getValueParameter();
+						if (value != null) {
+							String projectUri = template.getProjectUri();
+							ResolvedJavaTypeInfo whenSwitchJavaTypeInfo = javaCache.resolveJavaType(value, projectUri)
+									.getNow(null);
+							if (whenSwitchJavaTypeInfo != null) {
+								JavaTypeFilter filter = javaCache.getJavaTypeFilter(projectUri, nativeImagesSettings);
+								JavaTypeAccessibiltyRule javaTypeAccessibility = filter.getJavaTypeAccessibility(
+										whenSwitchJavaTypeInfo, template.getJavaTypesSupportedInNativeMode());
+								fillCompletionFields(whenSwitchJavaTypeInfo, javaTypeAccessibility, filter, range, projectUri,
+										existingVars, list);
+								fillCompletionMethods(whenSwitchJavaTypeInfo, javaTypeAccessibility, filter, range,

```

Comment:
You need to remove  this line code, only ENUM fields should be availabe:

![image](https://user-images.githubusercontent.com/1932211/186521929-61376bbb-5765-4079-9c5c-fc59b9ae1015.png)